framework/src/modules/auth/utils.ts

```diff
@@ -190,3 +190,25 @@ export const verifyNonce = (
 		status: transaction.nonce > senderAccount.nonce ? VerifyStatus.PENDING : VerifyStatus.OK,
 	};
 };
+
+export const getTransactionFromParameter = (transactionParameter: unknown) => {
+	if (!transactionParameter) {
+		throw new Error('A transaction object must be provided.');
+	}
+
+	if (typeof transactionParameter !== 'string') {
+		throw new Error('Transaction parameter must be a string.');
+	}
+
+	let transactionParameterObject;
+	try {
+		transactionParameterObject = JSON.parse(transactionParameter) as Record<string, unknown>;
+	} catch {
+		throw new Error('Transaction parameter is not a valid JSON object.');
+	}
```

Comment:
In the current implementation, we expect transaction from endpoint to be encoded transaction in hex string.
I think for now we should keep it consistent. ie) you can do something like below
```
- check input is hex string
- Create buffer from the hex string
- Transaction.fromBytes()
```

Reply:
done, please check.